### ISSUE TITLE
Fn::Split support?

### ISSUE THREAD

**Issue by omar-dulaimi** [serverless/typescript#38]
Hello,

I'm getting this validation error locally in my serverless.ts file:

```
Type '{ 'Fn::Split': string[]; }' is not assignable to type 'AwsCfInstruction[]'.
  Object literal may only specify known properties, and ''Fn::Split'' does not exist in type 'AwsCfInstruction[]'.ts(2322)
index.d.ts(1331, 3): The expected type comes from property 'securityGroupIds' which is declared here on type 'AwsLambdaVpcConfig'
```

So I looked into the library code and couldn't find it mentioned anywhere. I saw the others though, like Fn::Join and the rest.

This brings my question, is the split method supported? Because I checked the later releases after v2.29.0 commit by commit and nothing was there(A change log would be really great by the way).

**Comment by omar-dulaimi** [comment]
Okay, So I ended up doing this: 

```
        vpc: ({
            securityGroupIds: {
                'Fn::Split': [',', '${env:SECURITY_GROUP_IDS}'],
            },
            subnetIds: {
                'Fn::Split': [',', '${env:SUBNET_IDS}'],
            },
        } as unknown) as AwsLambdaVpcConfig,
```

But I get this warning while deploying:

```
Serverless: Configuration warning:
Serverless:   at 'provider.vpc.securityGroupIds': should be array
Serverless:   at 'provider.vpc.subnetIds': should be array
```

**Comment by fredericbarthelet** [comment]
Hi @omar-dulaimi and thanks for reporting your issue.
This repository aims to contains only TypeScript definitions compiled from Serverless core framework internal validation logic.
If you ignore the TS error showing that using `Fn::Split` is not a valid property for `securityGroupIds`, you'd  still have the framework complaining about it (hence the warning you got on your second  try).

I opened a PR aiming at allowing such use of Fn::Split, but I'm not sure wether additional development should be done in https://github.com/serverless/serverless/blob/dac06c8ce644f0d219329474cb83c644ca61d880/lib/plugins/aws/deployFunction.js#L283-L289 and https://github.com/serverless/serverless/blob/dac06c8ce644f0d219329474cb83c644ca61d880/lib/plugins/aws/package/compile/functions.js#L364-L376

**Comment by pveller** [comment]
Not a directly related question but I would appreciate your guidance. Now that type definitions are auto-generated, certain things are much clearer and certain things are not. Please take a look at `stream` event definition:

I do this:
```
      events: [
        {
          stream: {
            arn: {
              'Fn::GetAtt': ['MyTable', 'StreamArm'],
            },
            batchSize: 1,
          },
        },
      ],
```
and serverless fails fast on  `functions.propagate.events[0].stream.arn': should be string` with `configValidationMode: 'error'`. If I turn off `configValidationMode`, serverless package fails on `EventSourceArn.split is not a function`. This has been talked about long time ago here https://github.com/serverless/serverless/issues/2365 and later fixed here https://github.com/serverless/serverless/pull/3111.

The fix was simple - adding an explicit `type` attribute to the stream event definition. DefinitelyTyped has it (https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/serverless/plugins/aws/provider/awsProvider.d.ts#L408):

```
    interface Stream {
        arn: string | { [key: string]: any };
        batchSize?: number | string;
        startingPosition?: number | string;
        enabled?: boolean;
        type?: 'dynamodb' | 'kinesis';
    }
```

but  it's lost in auto-generated types:

```
        {
            stream:
              | AwsArnString
              | (
                  | {
                      arn: AwsCfFunction;
                      [k: string]: unknown;
                    }
                  | {
                      arn: AwsArnString;
                      [k: string]: unknown;
                    }
                );
          }
```
So my question is - **where to submit a PR or an Issue to get back those types attributes into the  generated type definitions?** Adding `type` solved my problem but it would be nice if it was visible on the type def and thus accessible via auto-complete

Thanks!

**Comment by omar-dulaimi** [comment]
Hello @fredericbarthelet and thank you for getting it merged.

I updated the library and removed the old unnecessary casting in my code, and now ended up with these warnings that I'm not sure if related to this issue.

`Type 'string' is not assignable to type 'AwsCfFunction'.ts(2322)
`

![Screenshot from 2021-04-30 11-06-57](https://user-images.githubusercontent.com/11743389/116666910-355dc300-a9a4-11eb-9799-835be9dc2e0c.png)

The binding above is due to our use of serverless dotenv plugin.
Please let me know if I should open a separate issue.
